connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -859,6 +865,49 @@ object SparkSession extends Logging {
     }
   }
 
+  /**
+   * Create a new Spark Connect server to connect locally.
+   */
+  private[sql] def withLocalConnectServer[T](f: => T): T = {
+    synchronized {
+      val remoteString = initOptions
+        .get("spark.remote")
+        .orElse(Option(System.getProperty("spark.remote"))) // Set from Spark Submit
+        .orElse(sys.env.get(SparkConnectClient.SPARK_REMOTE))
```

Comment:
Can you check how this interacts with `SparkConnectClient.builder().loadFromEnvironment()`. You seem to be setting stuff here that that function is trying to read.

Reply:
Yeah, they are basically more and less same.

- Here, it gets `remoteString`, and below, check if they are `local[...]` or `local-cluster[...]`.
- If so, it starts a Spark Connect server by either `local[...]` or `local-cluster[...]`
- Then sets `spark.remote` to `sc://localhost"` in order to create a Spark Connect client.